backend/lib/edgehog/devices/device/manual_actions/update_application.ex

```diff
@@ -0,0 +1,53 @@
+#
+# This file is part of Edgehog.
+#
+# Copyright 2024 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+defmodule Edgehog.Devices.Device.ManualActions.UpdateApplication do
+  @moduledoc false
+  use Ash.Resource.ManualUpdate
+
+  alias Edgehog.Astarte.Device.DeploymentUpdate.RequestData
+  alias Edgehog.Containers.Deployment
+
+  @deployment_update Application.compile_env(
+                       :edgehog,
+                       :astarte_deployment_update_module,
+                       Edgehog.Astarte.Device.DeploymentUpdate
+                     )
+
+  @impl Ash.Resource.ManualUpdate
+  def update(changeset, _opts, _context) do
+    from = changeset.arguments.from
+    to = changeset.arguments.to
+
+    device = changeset.data
+
+    with {:ok, from} <- fetch_deployment(device, from),
```

Comment:
I think here is better to stuck with `Ash` intended behaviour and fetch arguments trough `Ash.Changeset` functions
```suggestion
    device = changeset.data

    with {:ok, from} <- Ash.Changeset.fetch_argument(changeset, :from),
         {:ok, to} <- Ash.Changeset.fetch_argument(changeset, :to),
         {:ok, from} <- fetch_deployment(device, from),
```